src/core/math/Matrix.js

```diff
@@ -53,392 +51,369 @@ class Matrix
-
-       if (delta < 0.00001)
-       {
-           transform.rotation = skewY;
-
-           if (a < 0 && d >= 0)
-           {
-               transform.rotation += (transform.rotation <= 0) ? Math.PI : -Math.PI;
-           }
-
-           transform.skew.x = transform.skew.y = 0;
-
-       }
-       else
-       {
-           transform.skew.x = skewX;
-           transform.skew.y = skewY;
-       }
-
-       // next set scale
-       transform.scale.x = Math.sqrt(a * a + b * b);
-       transform.scale.y = Math.sqrt(c * c + d * d);
-
-       // next set position
-       transform.position.x = this.tx;
-       transform.position.y = this.ty;
-
-       return transform;
-   }
-
-
-   /**
-    * Inverts this matrix
-    *
-    * @return {PIXI.Matrix} This matrix. Good for chaining method calls.
-    */
-   invert()
-   {
-       let a1 = this.a;
-       let b1 = this.b;
-       let c1 = this.c;
-       let d1 = this.d;
-       let tx1 = this.tx;
-       let n = a1*d1-b1*c1;
-
-       this.a = d1/n;
-       this.b = -b1/n;
-       this.c = -c1/n;
-       this.d = a1/n;
-       this.tx = (c1*this.ty-d1*tx1)/n;
-       this.ty = -(a1*this.ty-b1*tx1)/n;
-
-       return this;
-   }
-
-
-   /**
-    * Resets this Matix to an identity (default) matrix.
-    *
-    * @return {PIXI.Matrix} This matrix. Good for chaining method calls.
-    */
-   identity()
-   {
-       this.a = 1;
-       this.b = 0;
-       this.c = 0;
-       this.d = 1;
-       this.tx = 0;
-       this.ty = 0;
-
-       return this;
-   }
+    /**
+     * Creates a Matrix object based on the given array. The Element to Matrix mapping order is as follows:
+     *
+     * a = array[0]
+     * b = array[1]
+     * c = array[3]
+     * d = array[4]
+     * tx = array[2]
+     * ty = array[5]
+     *
+     * @param array {number[]} The array that the matrix will be populated from.
+     */
+    fromArray(array) {
+        this.a = array[0];
+        this.b = array[1];
+        this.c = array[3];
+        this.d = array[4];
+        this.tx = array[2];
+        this.ty = array[5];
+    }
+
+
+    /**
+     * sets the matrix properties
+     *
+     * @param {number} a
+     * @param {number} b
+     * @param {number} c
+     * @param {number} d
+     * @param {number} tx
+     * @param {number} ty
+     *
+     * @return {PIXI.Matrix} This matrix. Good for chaining method calls.
+     */
+    set(a, b, c, d, tx, ty) {
+        this.a = a;
+        this.b = b;
+        this.c = c;
+        this.d = d;
+        this.tx = tx;
+        this.ty = ty;
+
+        return this;
+    }
+
+
+    /**
+     * Creates an array from the current Matrix object.
+     *
+     * @param transpose {boolean} Whether we need to transpose the matrix or not
+     * @param [out=new Float32Array(9)] {Float32Array} If provided the array will be assigned to out
+     * @return {number[]} the newly created array which contains the matrix
+     */
+    toArray(transpose, out) {
```

Comment:
Curly braces on a new line. Basically there are a lot of these. I won't comment on all of them.